resources/validations/ui/src/context/browser/presenter/machines/schematron.ts

```diff
@@ -0,0 +1,195 @@
+import { derived, Statemachine, statemachine } from 'overmind';
+
+import type {
+  SchematronAssert,
+  FailedAssert,
+} from '../../../../use-cases/schematron';
+import { createValidatorMachine, ValidatorMachine } from './validator';
+
+export type Role = string;
+
+type States =
+  | {
+      current: 'UNLOADED';
+    }
+  | {
+      current: 'LOADED';
+    };
+
+type BaseState = {
+  assertionsById: {
+    [assertionId: string]: SchematronAssert;
+  };
+  assertionGroups: {
+    title: string;
+    asserts: {
+      id: string;
+    }[];
+  }[];
+  filter: {
+    role: Role;
+    text: string;
+  };
+  filterRoles: Role[];
+  roles: Role[];
+  schematronReport: {
+    summaryText: string;
+    groups: {
+      title: string;
+      //see: string;
+      assertions: {
+        summary: string;
+        summaryColor: 'red' | 'green';
+        assertions: (SchematronAssert & {
+          icon: typeof checkCircleIcon;
+          fired: FailedAssert[];
+        })[];
+      };
+    }[];
+  };
+  schematronAsserts: SchematronAssert[];
+  schematronAssertsFiltered: SchematronAssert[];
+  validator: ValidatorMachine;
+};
+
+type Events = {
+  type: 'DATA_LOADED';
+  schematronAsserts: SchematronAssert[];
+};
+
+export type SchematronMachine = Statemachine<States, Events, BaseState>;
+
+const checkCircleIcon = { sprite: 'check_circle', color: 'green' };
+const helpIcon = { sprite: 'help', color: 'blue' };
+const cancelIcon = {
+  sprite: 'cancel',
+  color: 'red',
+};
+
+const schematronMachine = statemachine<States, Events, BaseState>({
+  UNLOADED: {
+    DATA_LOADED: schematronAsserts => {
+      return {
+        current: 'LOADED',
+        schematronAsserts,
+      };
+    },
+  },
+  LOADED: {},
+});
+
+export const createSchematronMachine = () => {
+  return schematronMachine.create(
+    { current: 'LOADED' },
+    {
+      assertionsById: derived((state: SchematronMachine) => {
+        const assertions: SchematronMachine['assertionsById'] = {};
+        state.schematronAssertsFiltered.forEach(assert => {
+          assertions[assert.id] = assert;
+        });
+        return assertions;
+      }),
+      assertionGroups: derived((state: SchematronMachine) => {
+        // Return a sample assertion group corresponding to the source rules.
+        return [
+          {
+            title: 'ssp.sch assertions',
```

Comment:
B: I think we should aim for something a little more user friendly because the FR reviewers won't know what this is.

Reply:
This is definitely just a placeholder label - but certainly I'll put something else in.